vcx/libvcx/src/utils/libindy/anoncreds.rs

```diff
@@ -85,11 +87,25 @@ pub fn libindy_prover_create_proof(proof_req_json: &str,
         .map_err(map_rust_indy_sdk_error_code)
 }
 
-pub fn libindy_prover_get_credentials_for_proof_req(proof_req: &str) -> Result<String, u32> {
+fn fetch_credentials(search_handle: i32, count: usize) -> Result<String, u32> {
+    Prover::fetch_credentials(search_handle, count).map_err(map_rust_indy_sdk_error_code)
+}
 
-    Prover::get_credentials_for_proof_req(get_wallet_handle(),
-                                          proof_req)
-        .map_err(map_rust_indy_sdk_error_code)
+pub fn libindy_prover_get_credentials_for_proof_req(proof_req: &str) -> Result<String, u32> {
+    let wallet_handle = get_wallet_handle();
+    let proof_request_json:Map<String, Value> = serde_json::from_str(proof_req).map_err(|_| INVALID_PROOF_REQUEST.code_num)?;
+    let (search_handle, count) = Prover::search_credentials(wallet_handle, Some("{}")).map_err(map_rust_indy_sdk_error_code)?;
+
+    // since the search_credentials_for_proof request validates that the proof_req is properly structured, this get()
+    // fn should never fail, unless libindy changes their formats.
+    let v = proof_request_json.get(REQUESTED_ATTRIBUTES).ok_or(INVALID_ATTRIBUTES_STRUCTURE.code_num)?;
+
+    let requested_attributes: Map<String, Value> = serde_json::from_value(v.clone())
+        .map_err(|_| { error!("Invalid Json Parsing of Requested Attributes Retrieved From Libindy. Did Libindy change its structure?"); INVALID_CREDENTIAL_JSON.code_num})?;
```

Comment:
Any references to the proof request could be removed. 1. proof_req (param), 2. proof_request_json, 3. requested_attributes